test/Garnet.test/RespListGarnetClientTests.cs

```diff
@@ -0,0 +1,140 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+using System.Threading;
+using System.Threading.Tasks;
+using Garnet.client;
+using NUnit.Framework;
+
+namespace Garnet.test
+{
+    [TestFixture]
+    public class RespListGarnetClientTests
+    {
+        private GarnetServer _server;
+
+        [OneTimeSetUp]
+        public void Setup()
+        {
+            TestUtils.DeleteDirectory(TestUtils.MethodTestDir, wait: true);
+            _server = TestUtils.CreateGarnetServer(TestUtils.MethodTestDir, lowMemory: true);
+            _server.Start();
+        }
+
+        [Test]
+        public void AddElementsToTheListHead_WithCallback()
```

Comment:
Looks like we're only testing for the validity of the count returned, would be good to also test validity of the content of the list (validate that items were inserted in the correct order and to the correct place in the list)

Reply:
This could be done if we implement `LRANGE`

Reply:
Well you can technically use a different client until that's implemented...

Reply:
Could you please elaborate?

Reply:
I mean you can use SE.Redis / LightClient to make these queries that are not yet supported by GarnetClient, but it's fine not to do that right away